packages/image_picker/image_picker_platform_interface/lib/src/method_channel/method_channel_image_picker.dart

```diff
@@ -72,13 +73,18 @@ class MethodChannelImagePicker extends ImagePickerPlatform {
       throw ArgumentError.value(maxHeight, 'maxHeight', 'cannot be negative');
     }
 
+    if (limit != null && limit < 2) {
```

Comment:
Why is this being re-asserted at a second layer?

Reply:
I've done it the same way as for the ImageOptions params. They are also double checked

Reply:
That seems likely to have been a mistake created during refactoring to create the platform interface layer. We don't need to assert things when both creating *and* reading from the options.

Reply:
Ok, I removed the limit assertion. I left image options assertions because removing them will break the tests for deprecated methods that returning `PickedFile`, that should be removed while fixing: https://github.com/flutter/flutter/issues/144286